### DISCUSSION TITLE
Trying to serialize number as octal string

### DISCUSSION THREAD

**Discussion by olensmar** [eemeli/yaml#465] [General]
Hi there, 

I have a js object with a number that i'm trying to serialize as an octal (0-prefixed) string - i've added a reviver function:

```
      let yaml = stringify(myObject, (key, value) => {
        return value && (key === 'defaultMode' || key === 'mode') ? value.toString().padStart(4, '0') : value;
      })
```

unfortunately the value comes out surrounded by quotes:

```
            defaultMode: "0644"
```

setting `blockQuote` to false didn't help - any suggestions on how I could get the above serialized as simply

```
            defaultMode: 0644
```

thanks in advance!

/Ole

**Comment by eemeli** [comment]
```js
import { Document } from 'yaml'

const doc = new Document({ defaultMode: 0644 })
doc.get('defaultMode', true).format = 'OCT'
String(doc)
```

```yaml
defaultMode: 0o644
```

That's using the default YAML 1.2 `core` schema. The octal representation would be `0644` if you build the doc with `schema: 'yaml-1.1'` as an option, so make sure to select whatever matches the consumer of your YAML.

**Reply by olensmar** [reply]
Thanks @eemeli - when I try this the value comes out with

`defaultMode: 0o1204`

here is my entire code block (I'm processing kubernetes deployments - hence the logic to find the nested `defaultMode`)

```
      let doc = new Document(data, {schema: 'yaml-1.1'});
      // @ts-ignore
      let volumes = doc.get('spec')?.get('template')?.get('spec')?.get('volumes');
      if (volumes?.items) {
        volumes.items.forEach((v: any) => {
          let defaultMode = v.get('secret')?.get('defaultMode', true);
          if (defaultMode) {
            defaultMode.format = 'OCT';
          }
        });
      }

      console.log( stringify(doc))
```

**Reply by eemeli** [reply]
You should call `String(doc)` or `doc.toString()` rather than `stringify(doc)`.

**Reply by olensmar** [reply]
thanks - almost there - now I have to reparse the value as an octal string:

```
            defaultMode.format = 'OCT';
            defaultMode.value = parseInt(defaultMode.value.toString(), 8);
```

but then it comes out without the leading 0 

```
            defaultMode: 644
```

the original data is parsed from yaml using parseAllDocuments and the document is converted to an object using toJS() - should I be doing it differently!?

thanks again for your help..

**Reply by olensmar** [reply]
my bad - I figured it out - I had to specify` {schema: 'yaml-1.1'} `when parsing the YAML also - thank you!

**Comment by olensmar** [comment]
follow-up question: I still want to pass some schemaOptions to the "stringification" - how do I do that with the doc.toString() method?

**Reply by olensmar** [reply]
never mind.. seems I could just pass them to toString - duh..

**Comment by olensmar** [comment]
Thanks for helping with this @eemeli - all working now!
